doc/dds-core-xtypes/README.md

```diff
@@ -235,51 +236,53 @@ node.access().member() // member used for access from parent to this node
 ```
 Note: `node.data()` will return a `ReadableDynamicDataRef` or a `WritableDynamicDataRef` depending of the node type.
 
-In order to break the tree iteration, the user call throw a boolean value.
+In order to break the tree iteration, the user can throw a boolean value.
+
 For example:
 ```
 case TypeKind::FLOAT_128_TYPE: // The user app does not support 128 float values
     throw false; // Break the tree iteration
 ```
 The boolean exception value will be returned by `for_each` function call.
-If the visitor callback implementation does not call an exception, `for_each` function will return `true` by default.
+If the visitor callback implementation does not call an exception, `for_each` function returns `true` by default.
 
 ## Performance
-The `DynamicData` instance only uses the minimal allocations needed to store the data.
+The `DynamicData` instance uses the minimal allocations needed to store any data.
 If its associated type only contains the following (nested) types: `PrimitiveType`, `StructType`, or `ArrayType`;
-the memory required can be got with only **one allocation** at the creation phase of the `DynamicData`.
-Only if the type contains some `MutableCollectionType`, another allocation will be performed to manage the inner types of that mutable type.
+the memory required can be allocated with only **one allocation** during the creation phase of the `DynamicData`.
+On the other hand, if the type contains any `MutableCollectionType`, other allocations will be performed in order to manage that inner types' mutable members.
```

Comment:
"other allocations" seems like a lot of it. Other way to say it?
The last word "members", it is not necessary to have members.

Reply:
Well, you do 1 allocation for each inner mutable type.  we could go with
[...] the minimum number or allocations [...]

about members: it is intended to be the noun rather than the actual class Member.